src/components/ec-radio-btn/ec-radio-btn.vue

```diff
@@ -0,0 +1,306 @@
+      @focus="setFocusState(true)"
+      @blur="setFocusState(false)"
+    >
+
+    <div
+      class="ec-radio-btn__wrapper"
+      :class="{
+        'ec-radio-btn__wrapper--is-single-line': isTextInline,
+      }"
+    >
+      <div
+        data-test="ec-radio-btn__icon-wrapper"
+        class="ec-radio-btn__icon-wrapper"
+        :class="{
+          'ec-radio-btn__icon-wrapper--focused': isFocused && !isChecked,
+          'ec-radio-btn__icon-wrapper--checked': isChecked && !isDisabled && !isFocused,
+          'ec-radio-btn__icon-wrapper--checked-and-focused': isChecked && isFocused,
+          'ec-radio-btn__icon-wrapper--error': hasError && !isChecked,
+          'ec-radio-btn__icon-wrapper--disabled': isDisabled && !isChecked,
+          'ec-radio-btn__icon-wrapper--checked-and-disabled': isDisabled && isChecked,
+        }"
+      >
+        <div
+          class="ec-radio-btn__icon-wrapper-inner"
+        >
+
+          <ec-icon
+            class="ec-radio-btn__icon"
+            :class="{
+              'ec-radio-btn__icon--checked': isChecked && !isDisabled,
+              'ec-radio-btn__icon--checked-and-disabled': isDisabled && isChecked
+            }"
+            name="rounded-notification"
+            :size="20"
+          />
+        </div>
+      </div>
+
+      <div
+        v-if="hasLabel"
+        data-test="ec-radio-btn__text-wrapper"
+        class="ec-radio-btn__text-wrapper"
+        :class="{
+          'ec-radio-btn__text-wrapper--is-single-line': isTextInline,
+        }"
+      >
+        <label
+          :for="name"
+          class="ec-radio-btn__label"
+          :class="{
+            'ec-radio-btn__label--disabled': isDisabled,
+          }"
+          :title="isTextInline ? label : undefined"
+          data-test="ec-radio-btn__label"
+        >
+          <slot name="label">{{ label }}</slot>
+        </label>
+
+        <p
+          v-if="hasDescription"
+          :for="id"
+          class="ec-radio-btn__description"
+          :class="{
+            'ec-radio-btn__description--is-single-line': isTextInline,
+            'ec-radio-btn__description--disabled': isDisabled,
+          }"
+          :title="isTextInline ? description : undefined"
+          data-test="ec-radio-btn__description"
+        >
+          <slot name="description">{{ description }}</slot>
+        </p>
+      </div>
+    </div>
+
+    <div
+      :id="errorId"
+      v-if="hasErrorMessage"
+      class="ec-radio-btn__error-text"
+      data-test="ec-radio-btn__error-text"
+    >
+      <slot name="error-message">{{ errorMessage }}</slot>
+    </div>
+  </div>
+
+</template>
+
+<script setup lang="ts">
+import {
+  computed, ref, useSlots, withDefaults,
+} from 'vue';
+
+import { getUid } from '../../utils/uid';
+import EcIcon from '../ec-icon';
+import type { RadioButtonEvents } from './types';
+import { RadioButtonEvent } from './types';
+
+interface RadioButtonProps {
+  value: string,
+  modelValue?: string,
```

Comment:
What is the difference between these props? Why there are two value props? Why one is required and another is not?

Reply:
one value is the one belonging to the option. the model value instead is the v-model of the input (https://vuejs.org/guide/essentials/forms.html#radio)